### ISSUE TITLE
`unidiomatic-typecheck` not flagged for type(x) is type(y)

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#10365]
The spirit of `unidiomatic-typecheck` is to flag unidiomatic typechecks.

This is unidiomatic:
```
type(x) is type(y)
```

It should be one of:
```
isinstance(x, type(y))
issubclass(type(x), type(y))
```

**Comment by jacobtylerwalls** [comment]
Exactly. The spirit of the message as I understand it is to discourage that kind of exact check.

**Comment by DanielNoord** [comment]
I have some use cases in my codebases where I need this, but would personally be fine with having to disable the check there. This is likely a case where we need input from the primer to determine how good/bad it would be?

**Comment by Pierre-Sassoulas** [comment]
The linked MR with a fix already exists and does not have a very good primer imo, see https://github.com/pylint-dev/pylint/pull/10372#issuecomment-2854435684. I think this is an opinionated check and my intuition is that it's been toned down in the past.  I'm currently git blaming the code to see the reasoning for the way the code is right now and have a more informed opinion. (made the comment to point out that the fix/primer already exists)/

**Comment by jacobtylerwalls** [comment]
We should add an exception for uses inside `__eq__(self, other)`. A lot of the primer cases are coming from traffic like that.
